### PR TITLE
Adds back `NavigationRestrictions` that removes navigability for `driveItem/workbook` navigation property

### PR DESCRIPTION
**Pull request by irvinesunday**
Fixes https://github.com/microsoftgraph/msgraph-metadata/issues/270

This PR:
- Removes the navigability for `driveItem/workbook` navigation property for the target CSDL file - `cleanMetadataWithDescriptionsAndAnnotations` (v1.0 and beta)

### CODE REVIEW

File: transforms/csdl/preprocess_csdl.xsl [hunk 1]
```diff
@@ -786,6 +786,25 @@
     <xsl:template match="edm:Schema[@Namespace='microsoft.graph']">
         <xsl:copy>
             <xsl:apply-templates select="@* | node()"/>
+            
+            <!-- Remove navigability for driveItem/workbook navigation property for DevX API-specific CSDL-->
+            <xsl:choose>
+                <xsl:when test="$remove-capability-annotations='False'">
```
Comment (baywet) [inline transforms/csdl/preprocess_csdl.xsl:792]: assuming this is true for kiota based SDKs?

Comment (andrueastman) [reply transforms/csdl/preprocess_csdl.xsl:792]: OpenApi generation for Kiota based SDKs does include annotations so this navigability will also be removed for Kiota based SDKs. 
To omit the file for Kiota based SDKs we may need to include the `open-api-generation` variable in the test [here ](https://github.com/microsoftgraph/msgraph-metadata/blob/master/transforms/csdl/preprocess_csdl.xsl#L11)

Comment (baywet) [reply transforms/csdl/preprocess_csdl.xsl:792]: Thanks for the context. I believe kiota deceptions should not have the restriction. Irvine, can you make the change please?

Comment (irvinesunday) [reply transforms/csdl/preprocess_csdl.xsl]: This is effected.

**Review by baywet — APPROVED** [review]

**Pull request merged**